homeassistant/components/ness_alarm/config_flow.py

```diff
@@ -0,0 +1,235 @@
+
+    Args:
+        hass (HomeAssistant): The Home Assistant instance.
+        data (dict[str, Any]): Dictionary containing user input, must include
+            CONF_HOST (str): Hostname or IP address of the Ness Alarm device.
+            CONF_PORT (int): Port number to connect to.
+
+    Returns:
+        dict[str, Any]: Dictionary with a "title" key describing the Ness Alarm device.
+
+    Raises:
+        NessAlarmConnectionError: If unable to connect to the specified host and port.
+    """
+    host = data[CONF_HOST]
+    port = data[CONF_PORT]
+
+    def test_connection():
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.settimeout(CONNECTION_TIMEOUT_SECONDS)
+        try:
+            result = sock.connect_ex((host, port))
+        except OSError:
+            return False
+        else:
+            return result == 0
+        finally:
+            sock.close()
+
+    is_connected = await hass.async_add_executor_job(test_connection)
+
+    if not is_connected:
+        raise NessAlarmConnectionError(f"Cannot connect to {host}:{port}")
+
+    return {"title": f"Ness Alarm ({host})"}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Ness Alarm."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> config_entries.ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            await self.async_set_unique_id(
+                f"{user_input[CONF_HOST]}:{user_input.get(CONF_PORT, DEFAULT_PORT)}"
+            )
+            self._abort_if_unique_id_configured()
+
+            try:
+                info = await validate_input(self.hass, user_input)
+            except NessAlarmConnectionError:
+                errors["base"] = "cannot_connect"
+            except Exception:
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                return self.async_create_entry(title=info["title"], data=user_input)
+
+        schema = vol.Schema(
+            {
+                vol.Required(CONF_HOST): str,
+                vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
+                vol.Required(
+                    CONF_MAX_SUPPORTED_ZONES, default=DEFAULT_MAX_SUPPORTED_ZONES
+                ): int,
+                vol.Optional(
+                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+                ): vol.All(cv.positive_int, vol.Range(min=1, max=3600)),
+                vol.Optional(
+                    CONF_INFER_ARMING_STATE, default=DEFAULT_INFER_ARMING_STATE
+                ): bool,
+                vol.Optional(
+                    CONF_SUPPORT_HOME_ARM, default=DEFAULT_SUPPORT_HOME_ARM
+                ): bool,
+            }
+        )
```

Comment:
Only the things related to the connection should be part of the config flow, the rest can be in the options flow. I would assume `max_supported_zones` belongs here as well